auth/src/main/java/io/grpc/auth/GoogleAuthLibraryCallCredentials.java

```diff
@@ -353,4 +357,30 @@ public Credentials tryServiceAccountToJwt(Credentials creds) {
       return creds;
     }
   }
+
+  /**
+   * This method is to support the hack for AppEngineCredentials which need to run on a
+   * specific thread.
+   * @return Whether a specific executor is needed or if any executor can be used
+   */
+  @Override
+  public boolean isSpecificExecutorRequired() {
+    // Cache the value so we only need to try to load the class once
+    if (requiresSpecificExecutor == null) {
+      if (googleCredentialsClass == null) {
```

Comment:
Why is `googleCredentialsClass` involved? We need to be checking whether `creds` is AppEngineCredentials.

Reply:
You're right, done.